### ISSUE TITLE
Add an iOS .app build  to run UI e2e automatic tests on iOS simulators

### ISSUE THREAD

**Issue by hetvart** [status-im/status-react#6682]
[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a QA, I want to have both Android and iOS application builds to run e2e tests against every PR and nightly. 
### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Infrastructure Issue

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)
*Summary*: 
We use [SauceLabs](saucelabs.com) cloud to perform UI e2e tests with predefined VMs (simulators).
In order to run the same set of tests on iOS we need to build **_.app_** file for installing it on iOS simulators.
Requirements for SauceLabs iOS Mobile Application Testing are given [here](https://wiki.saucelabs.com/display/DOCS/Requirements+for+Android+and+iOS+Mobile+Application+Testing#RequirementsforAndroidandiOSMobileApplicationTesting-iOSRequirements).

Also, we need to have this **_.app_** build artifact in [nightly pipeline](https://ci.status.im/job/status-react/job/nightly/).

**Comment by hetvart** [comment]
@jakubgs 
could you help me with this?

**Comment by jakubgs** [comment]
Sure, so we'll need another parallel build for iOS for e2e tests like we have for Android right?
I'll take a look at the requirements you linked tomorrow.

**Comment by hetvart** [comment]
@jakubgs 
yes, it should build .app file which we can install on iOS simulator then.
thank you!

**Comment by jakubgs** [comment]
don't you mean an `*.ipa` file?

**Comment by hetvart** [comment]
no, there are two type of builds: for real devices and simulators.
For a real device it needs to be built with a development identity and provisioning profile, so we use `*.ipa` on real devices.
For a simulator it needs to be a simulator build of the application i.e. `*.app`

**Comment by jakubgs** [comment]
Ah, okay, thanks for clearing that up.

**Comment by jakubgs** [comment]
Damn, I was too busy today to get around to this, sorry man. I'll make sure to start work on this on Monday.

**Comment by hetvart** [comment]
thank you!

**Comment by jakubgs** [comment]
Question: Do we need speicfically an `*.app` file inside of a zip, or would an `*.app.dSYM` file in a zip archive also work?

**Comment by jakubgs** [comment]
Also not sure what's the difference between an `*.app` and an `*.app *.xcarchive`, example:
```
StatusIm.app 2018-11-21 18.10.47.xcarchive
```

**Comment by jakubgs** [comment]
Oh I see, the `xarchive` contains the `app` under `Products/Applications`:
```
administrators-Mac-Pro:status-e2e administrator$ ls -l
total 0
drwxr-xr-x@ 6 jenkins  staff  192 Nov 21 18:18 StatusIm.app 2018-11-21 18.10.47.xcarchive
administrators-Mac-Pro:status-e2e administrator$ ls -l StatusIm.app\ 2018-11-21\ 18.10.47.xcarchive/
total 8
drwxr-xr-x  6 jenkins  staff  192 Nov 21 18:18 BCSymbolMaps
-rw-r--r--  1 jenkins  staff  819 Nov 21 18:18 Info.plist
drwxr-xr-x  3 jenkins  staff   96 Nov 21 18:13 Products
drwxr-xr-x  5 jenkins  staff  160 Nov 21 18:18 dSYMs
administrators-Mac-Pro:status-e2e administrator$ ls -l StatusIm.app\ 2018-11-21\ 18.10.47.xcarchive/Products/Applications/
total 0
drwxr-xr-x  19 jenkins  staff  608 Nov 21 18:18 StatusIm.app
administrators-Mac-Pro:status-e2e administrator$ ls -l StatusIm.app\ 2018-11-21\ 18.10.47.xcarchive/Products/Applications/StatusIm.app/
total 90240
-rw-r--r--  1 jenkins  staff      1710 Nov 21 18:13 AppIcon60x60@2x.png
-rw-r--r--  1 jenkins  staff      2176 Nov 21 18:13 AppIcon76x76@2x~ipad.png
-rw-r--r--  1 jenkins  staff    185144 Nov 21 18:13 Assets.car
drwxr-xr-x  4 jenkins  staff       128 Nov 21 18:18 Frameworks
-rw-r--r--  1 jenkins  staff       962 Nov 21 18:13 GoogleService-Info.plist
-rw-r--r--  1 jenkins  staff      2570 Nov 21 18:13 Info.plist
-rw-r--r--  1 jenkins  staff    224384 Nov 21 18:13 Inter-UI-Bold.otf
-rw-r--r--  1 jenkins  staff    222872 Nov 21 18:13 Inter-UI-Medium.otf
-rw-r--r--  1 jenkins  staff    215120 Nov 21 18:13 Inter-UI-Regular.otf
-rw-r--r--  1 jenkins  staff         8 Nov 21 18:13 PkgInfo
-rwxr-xr-x  1 jenkins  staff  33355136 Nov 21 18:18 StatusIm
drwxr-xr-x  3 jenkins  staff        96 Nov 21 18:18 _CodeSignature
drwxr-xr-x  4 jenkins  staff       128 Nov 21 18:18 assets
-rw-r--r--  1 jenkins  staff     12129 Nov 21 18:18 embedded.mobileprovision
drwxr-xr-x  5 jenkins  staff       160 Nov 21 18:13 launch-image-universal.storyboardc
-rw-r--r--  1 jenkins  staff  11871377 Nov 21 18:18 main.jsbundle
-rw-r--r--  1 jenkins  staff     79314 Nov 21 18:13 message.wa
```

**Comment by jakubgs** [comment]
Another thing I'm not sure of this should be build with `export_method` set to `ad-hoc` as we do for all our dev builds, or `development`

**Comment by jakubgs** [comment]
I did find this:
https://github.com/saucelabs/ios-rdc-quickstart
And it states:
>Sauce Labs can accept your app in two formats.
>* As an IPA archive, whose filename must end in .ipa; or
>* A zipped .app directory, whose filename must end in .zip.

Not sure how correct that is - it is 3 years old - but if that's the case then we don't actually need any special build, just our regular `*.ipa` files produced by current builds would work.

So now I'm confused, do we even need a separate build to make a zip of `*.app` or not?

**Comment by hetvart** [comment]
> I did find this:
> https://github.com/saucelabs/ios-rdc-quickstart
> And it states:
> 
> > Sauce Labs can accept your app in two formats.
> > 
> > * As an IPA archive, whose filename must end in .ipa; or
> > * A zipped .app directory, whose filename must end in .zip.
> 
> Not sure how correct that is - it is 3 years old - but if that's the case then we don't actually need any special build, just our regular `*.ipa` files produced by current builds would work.
> 
> So now I'm confused, do we even need a separate build to make a zip of `*.app` or not?

Definitely, we do need it. The main difference between `*.ipa` and `*.app` is that you cannot install `*.ipa` on iOS emulator. We run e2e tests on cloud emulators via Appium server. The iOS simulator app is a different compiled binary than the app that gets published on the app store (which requires personal apple developer id to sign/certify it etc.).

Here is official Appium info on that:
http://appium.io/docs/en/writing-running-appium/running-tests/#preparing-your-app-for-test-ios

And a bit info on how it works on real device:
http://appium.io/docs/en/drivers/ios-xcuitest-real-devices/

<blockquote><img src="https://avatars0.githubusercontent.com/u/88837?s=400&v=4" width="48" align="right"><div><img src="https://assets-cdn.github.com/favicon.ico" height="14"> GitHub</div><div><strong><a href="https://github.com/saucelabs/ios-rdc-quickstart">saucelabs/ios-rdc-quickstart</a></strong></div><div>How to get started with the iOS Real Device Cloud. - saucelabs/ios-rdc-quickstart</div></blockquote>
<blockquote><img src="/img/favicon.ico" width="48" align="right"><div><strong><a href="http://appium.io/docs/en/writing-running-appium/running-tests/#preparing-your-app-for-test-ios">Running Tests - Appium</a></strong></div></blockquote>
<blockquote><img src="/img/favicon.ico" width="48" align="right"><div><strong><a href="http://appium.io/docs/en/drivers/ios-xcuitest-real-devices/">XCUITest Real Devices (iOS) - Appium</a></strong></div></blockquote>

**Comment by jakubgs** [comment]
Okay, I've merged https://github.com/status-im/status-react/pull/6785, though this still needs some work. But at least we have a stable build that just needs some tweaks. I don't want to keep this PR open for too long while `develop` diverges.

**Comment by hetvart** [comment]
> Okay, I've merged #6785, though this still needs some work. But at least we have a stable build that just needs some tweaks. I don't want to keep this PR open for too long while `develop` diverges.

Great! Agree with you about merging to `develop`.
Thank you!
